contracts/ERC721A.sol

```diff
@@ -613,6 +613,22 @@ contract ERC721A is IERC721A {
         }
     }
 
+    /**
+     * @dev Returns whether `spender` is allowed to manage `tokenId`.
+     *
+     * Requirements:
+     *
+     * - `tokenId` must exist.
+     */
+    function _isApprovedOrOwner(address spender, uint256 tokenId) internal view virtual returns (bool) {
+        address owner = ownerOf(tokenId);
+        bool result = true;
+        if (spender != owner)
+            if (!isApprovedForAll(owner, spender))
+                if (getApproved(tokenId) != spender) result = false;
```

Comment:
I'm realizing now that we have both `getApproved` and `_getApprovedAddress` which are very similar. I'd like our code to be a bit more cohesive - can we cut this down to one function or at least call _getApprovedAddress in getApproved? Right now this feels fragmented.

Reply:
I can rename it to `_getApprovedAddressAndSlot`.

Reply:
`getApproved` is part of the ERC721 spec, and has the logic of ensuring the `tokenId` exists.

`_getApprovedAddress` is meant to return both the slot and the value of an approved address without checks. 

I don't think calling `_getApprovedAddress` in `getApproved` will do much to improve aesthetics,   
since it getting the value is simply a one-liner

```solidity
_tokenApprovals[tokenId].value
```

I'll rename `_getApprovedAddress` to `_getApprovedSlotAndAddress` to signify that the function has some low-level meaning.